### PR TITLE
Improve PHPStan and Psalm

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is where the static analysis tools are configured.

### CODE REVIEW

File: tests/Fixtures/Traits/ORM/Knplabs/Article.php [hunk 1]
```diff
@@ -25,9 +25,18 @@
...
+     * @ORM\Column(type="integer")
+     *
+     * @var int|null
+     */
+    public $id;
+
     /**
      * @psalm-var TranslationInterface[]|Collection
-     * @phpstan-var Collection<array-key, ArticleTranslation>
+     * @phpstan-var TranslationInterface[]|Collection<array-key, TranslationInterface>
```
Comment (VincentLanglet) [inline tests/Fixtures/Traits/ORM/Knplabs/Article.php:39]: Shouldnt this just be
```
@var Collection<array-key, TranslationInterface>
```
?

Comment (jordisala1991) [reply tests/Fixtures/Traits/ORM/Knplabs/Article.php:39]: If I do what  you suggest:

```
 ------ ------------------------------------------------------------------------------------------------------------------------------
  Line   tests/Fixtures/Traits/ORM/Knplabs/Article.php
 ------ ------------------------------------------------------------------------------------------------------------------------------
  48     PHPDoc type Doctrine\Common\Collections\Collection<(int|string), Knp\DoctrineBehaviors\Contract\Entity\TranslationInterface>
         of property Sonata\TranslationBundle\Tests\Fixtures\Traits\ORM\Knplabs\Article::$translations is not the same as PHPDoc type
         Doctrine\Common\Collections\Collection&iterable<Knp\DoctrineBehaviors\Contract\Entity\TranslationInterface> of overridden
         property Sonata\TranslationBundle\Tests\Fixtures\Model\Knplabs\TranslatableEntity::$translations.
         💡 You can fix 3rd party PHPDoc types with stub files:

            https://phpstan.org/user-guide/stub-files
            This error can be turned off by setting
            reportMaybesInPropertyPhpDocTypes: false in your phpstan.neon.dist.
 ------ ------------------------------------------------------------------------------------------------------------------------------
```

and psalm:

```
ERROR: NonInvariantDocblockPropertyType - tests/Fixtures/Traits/ORM/Knplabs/Article.php:48:15 - Property Sonata\TranslationBundle\Tests\Fixtures\Traits\ORM\Knplabs\Article::$translations has type Doctrine\Common\Collections\Collection<array-key, Knp\DoctrineBehaviors\Contract\Entity\TranslationInterface>, not invariant with Knp\DoctrineBehaviors\Model\Translatable\TranslatablePropertiesTrait::$translations of type Doctrine\Common\Collections\Collection|array<array-key, Knp\DoctrineBehaviors\Contract\Entity\TranslationInterface> (see https://psalm.dev/267)
    protected $translations;
```

Comment (VincentLanglet) [reply tests/Fixtures/Traits/ORM/Knplabs/Article.php:39]: I see, phpdoc need to be fixed here: https://github.com/KnpLabs/DoctrineBehaviors/blob/master/src/Model/Translatable/TranslatablePropertiesTrait.php

**Review by VincentLanglet — APPROVED** [review]

File: tests/Fixtures/Traits/ORM/Knplabs/Article.php (hunk 1 above)
Comment (VincentLanglet) [reply tests/Fixtures/Traits/ORM/Knplabs/Article.php:39]: I opened https://github.com/KnpLabs/DoctrineBehaviors/pull/671

Comment (jordisala1991) [reply tests/Fixtures/Traits/ORM/Knplabs/Article.php:39]: Can be improved later if that PR gets merged.

**Pull request merged**